virtwho/parser.py

```diff
@@ -506,22 +507,12 @@ def parse_options():
                 display_option = '%s-%s' % (cli_options['virtType'], option)
             elif option in SAT_OPTION_MAP:
                 display_option = SAT_OPTION_MAP[option]
-            logger.warning("The option --%s is deprecated and will be removed in the next release. "
-                  "Please see 'man virt-who-config' for details on adding a configuration section." % display_option)
-
-
-
-    for option in DEPRECATED_OPTIONS:
-        display_option = option
-        if option in cli_options and not cli_options[option] == defaults[option]:
-            if option == 'virtType' or option == 'smType':
-                display_option = cli_options[option]
-            elif any(option in s for s in VIRT_TYPE_OPTIONS):
-                display_option = '%s-%s' % (cli_options['virtType'], option)
-            elif option in SAT_OPTION_MAP:
-                display_option = SAT_OPTION_MAP[option]
-            logger.warning("The option --%s is deprecated and will be removed in the next release. "
-                  "Please see 'man virt-who-config' for details on adding a configuration section." % display_option)
+            used_deprecated_cli_options.append(display_option)
+    deprecated_options_msg = "The following cli options: %s are deprecated and will be removed " \
+    "in the next release. Please see 'man virt-who-config' for details on adding a configuration "\
+    "section."
+    if used_deprecated_cli_options:
+        logger.warning(deprecated_options_msg % ''.join(used_deprecated_cli_options))
```

Comment:
This code does not generate list of options. Following code:
```python
l = ['foo', 'bar', 'pub']
''.join(l)
```
will generate following string: "foobarpub" and not: "'--foo', '--bar', '--pub'". You probably want to use something like this:

```python
l = ['foo', 'bar', 'pub']
", ".join(['\'--' + item + '\'' for item in l])
```